client/tests/v2/tests.go

```diff
@@ -248,14 +260,31 @@ func testVarsToSecret(test *testsv2.Test, secret *corev1.Secret) {
 					},
 				},
 			}
+
+			test.Spec.Variables[k] = v
+		}
+	}
+
+	if len(snapshot) != 0 {
+		random, err := uuid.NewRandom()
+		if err != nil {
+			return err
+		}
+
+		data, err := json.Marshal(snapshot)
+		if err != nil {
+			return err
 		}
-		test.Spec.Variables[k] = v
+
+		secret.StringData[random.String()] = string(data)
```

Comment:
Why we need to fill two keys ?

Reply:
So, we will have a hash map with a history of secret vars map[uuid]secret vars. Need to have a reference for current uuid in order to pass it to execution